CMakeLists.txt

```diff
@@ -155,16 +156,47 @@ if (${H5_LIBVER_DIR} GREATER 16)
     #-----------------------------------------------------------------------------
     if (EXISTS "${H5EXAMPLES_SOURCE_DIR}/JAVA" AND IS_DIRECTORY "${H5EXAMPLES_SOURCE_DIR}/JAVA")
       option (HDF_BUILD_JAVA "Build JAVA support" OFF)
-      if (HDF_BUILD_JAVA AND HDF5_BUILD_JAVA)
-        add_subdirectory (JAVA)
-      endif ()
+    else ()
+      set (HDF_BUILD_JAVA OFF CACHE BOOL "Build examples JAVA support" FORCE)
     endif ()
+  else ()
+    set (HDF_BUILD_JAVA OFF CACHE BOOL "Build examples JAVA support" FORCE)
   endif ()
 
   #-----------------------------------------------------------------------------
   # Option to build filter examples
   #-----------------------------------------------------------------------------
-  if (EXISTS "${H5EXAMPLES_SOURCE_DIR}/C/H5F" AND IS_DIRECTORY "${H5EXAMPLES_SOURCE_DIR}/C/H5F")
+  if (EXISTS "${H5EXAMPLES_SOURCE_DIR}/C/H5FLT" AND IS_DIRECTORY "${H5EXAMPLES_SOURCE_DIR}/C/H5FLT")
     option (HDF_BUILD_FILTERS "Test filter support" OFF)
+    if (HDF_BUILD_FILTERS AND HDF5_ENABLE_PLUGIN_SUPPORT)
+      if(DEFINED ENV{HDF5_PLUGIN_PATH})
+        message (STATUS "ENV PATH=$ENV{HDF5_PLUGIN_PATH}")
+        set (H5EX_HDF5_PLUGIN_PATH $ENV{HDF5_PLUGIN_PATH})
+      else ()
+        message (STATUS "LIBRARY PATH=${HDF5_LIBRARY_PATH}/plugin")
+        set (H5EX_HDF5_PLUGIN_PATH ${HDF5_LIBRARY_PATH}/plugin)
+      endif ()
+      message (STATUS "H5EX_HDF5_PLUGIN_PATH=${H5EX_HDF5_PLUGIN_PATH}")
+    else ()
+      set (HDF_BUILD_FILTERS OFF CACHE BOOL "Build examples PLUGIN filter support" FORCE)
+    endif ()
+  else ()
+    set (HDF_BUILD_FILTERS OFF CACHE BOOL "Build examples PLUGIN filter support" FORCE)
   endif ()
+else ()
+  set (HDF_BUILD_FORTRAN OFF} CACHE BOOL "Build examples FORTRAN support" FORCE)
+  set (HDF_BUILD_JAVA OFF CACHE BOOL "Build examples JAVA support" FORCE)
+  set (HDF_BUILD_FILTERS OFF CACHE BOOL "Build examples PLUGIN filter support" FORCE)
 endif ()
+
+#-----------------------------------------------------------------------------
+# Build examples
+#-----------------------------------------------------------------------------
+add_subdirectory (C)
+if (HDF_BUILD_FORTRAN AND HDF5_BUILD_FORTRAN)
```

Comment:
It would be good to update these variable names to use a prefix like "H5EX" or "HDF5EX" as used above so that we know where the variables come from and what they're used for. It would also be good to decide which among "H5EX" and "HDF5EX" we wish to use and try to be consistent using that every across these CMakeLists files for the examples.